File: 프로그래머스/3/12987. 숫자 게임/숫자 게임.py
def solution(A, B):
    A.sort(reverse=True)
    B.sort(reverse=True)
    
    idx_a = 0
    idx_b = 0
    answer = 0
    for i in range(len(A)):
        if A[idx_a] < B[idx_b]:
            answer += 1
            idx_a += 1
            idx_b += 1
        else:
            idx_a += 1
    return answer

# 7 6 2 1
# 8 6 2 2

# 7 6 2 1
# 8 2 6 2

File: 프로그래머스/3/12987. 숫자 게임/숫자 게임.java
import java.util.*;

class Solution {
    public int solution(int[] A, int[] B) {
        Arrays.sort(A);
        Arrays.sort(B);
        
        int idxA = A.length - 1;
        int idxB = B.length - 1;
        int answer = 0;
        
        for (int i = 0; i < A.length; i++) {
            if (A[idxA] < B[idxB]) {
                answer++;
                idxA--;
                idxB--;
            } else {
                idxA--;
            }
        }
        return answer;
    }
}